timesketch/lib/utils.py

```diff
@@ -246,12 +246,75 @@ def read_and_validate_jsonl(file_handle):
     """
     # Fields that must be present in each entry of the JSONL file.
     mandatory_fields = ['message', 'datetime', 'timestamp_desc']
-    lineno = 0
+    line_number = 0
+    max_len_line_jsonl = current_app.config.get('MAX_LENGTH_LINE_JSONL')
+    tag_reducted = current_app.config.get('TAG_REDUCTED_JSONL')
     for line in file_handle:
-        lineno += 1
+        line_number += 1
         try:
             linedict = json.loads(line)
             ld_keys = linedict.keys()
+
+            current_len = len(line)
+            if current_len > max_len_line_jsonl:
+                field_len={}
+                total_len=0
+                for key in ld_keys:
+                    if isinstance(linedict[key], str):
+                        field_len[key]=len(linedict[key])
+                        total_len+=field_len[key]
+                    elif isinstance(linedict[key], list):
+                        field_len[key]=len(str(linedict[key]))
+                        total_len+=field_len[key]
+                field_len = dict(sorted(
+                        field_len.items(),
+                        key=lambda item: item[1],
+                        reverse=True)
+                    )
+                cur_len_fields = 0
+                reduc_field = []
```

Comment:
OK, I would like to see a comment here to explain a bit the logic here, since it seems to go into a bit more complicated calculations here... would be good for potential debugging sessions...

why are you dividing by 100 and multiplying by 10? why these subtractions and divisions, etc... can you explain the logic  better?

Reply:
Hi,
I tried to remember my idea!

https://github.com/lprat/timesketch/blob/proposal_issue_1754/timesketch/lib/utils.py#L290 : I search fields with long data (sorted), i put key in reduc_field, when lenght of the remaining fields  is smaller than max_lent_line_jsonl, I found all keys to reduce.

https://github.com/lprat/timesketch/blob/proposal_issue_1754/timesketch/lib/utils.py#L292 :
current_max: I take 90% of max_len_line_jsonl (for json structure not calculated in length). It may be too much ... Or unnecessary ...

https://github.com/lprat/timesketch/blob/proposal_issue_1754/timesketch/lib/utils.py#L295 : I am looking for the size that remains free (allocated to field to reduce)

https://github.com/lprat/timesketch/blob/proposal_issue_1754/timesketch/lib/utils.py#L296 : I get this size for each field to reduce.

After, I try to reduce fields in reduc_field[]